app/models/transaction.rb

```diff
@@ -16,45 +16,69 @@ class Transaction < ApplicationRecord
   )
   validates(
     :amount,
-    presence: true
+    presence: true,
+    numericality: { greater_than_or_equal_to: 0 }
   )
   validates(
     :transaction_type,
-    presence: true
+    presence: true,
+    inclusion: %w( NEFT Withdraw Deposite )
```

Comment:
@babarajay , Why it is a string and not a symbol?

Reply:
@babarajay , ?

Reply:
I actually don't understand your concert about your comment.

Reply:
@babarajay , `%i(neft withdraw deposite)` Please check the output of this.

Reply:
Sure

Reply:
It is not working for mycase; gives me validation error: "is not included in the list"

Reply:
@babarajay , Please solve that error.

Reply:
I searched about the difference between %i and %w but not able to got the difference; I would be great if you let me know the difference it has then I'll resolve it. Thanks

Reply:
Kind request to resolve my query. Thanks